### DISCUSSION TITLE
Can't install rugby with error

### DISCUSSION THREAD

**Discussion by kfmes** [swiftyfinch/Rugby#107] [Q&A] [answered]
Hi, i'm facing error on installing rugby 1.15.

attach log
```
[18/19] Compiling Rugby CacheHeader.swift
/private/var/folders/56/2w_2nmd16gx73g1cl_q4j5880000gp/T/mint/github.com_swiftyfinch_rugby/Sources/Rugby/Commands/Plan/Command/PlanRun.swift:15:32: error: variable used within its own initial value
        let selectedPlan = try selectedPlan(plans)
                               ^
/private/var/folders/56/2w_2nmd16gx73g1cl_q4j5880000gp/T/mint/github.com_swiftyfinch_rugby/Sources/Rugby/Commands/Plan/Command/PlanRun.swift:20:24: error: type of expression is ambiguous without more context
        let time = try measure {
                   ~~~~^~~~~~~~~
🌱 Encountered error during "swift build -c release -Xswiftc -target -Xswiftc x86_64-apple-macosx10.15". Use --verbose to see full output
🌱  Failed to build rugby 1.15.0 with SPM
```

install successfully at old version. 1.12.4 .

**Comment by swiftyfinch** [comment, marked as answer]
@kfmes Hello! Unfortunately, you need to use at least Xcode 12.5 CLT (Command Line Tool) for building 🏈 Rugby 1.15.0

**Reply by kfmes** [reply]
Oh I see that. I'm using Catalina, Catalina is supports up to xcode 12.4. (12.5  need to upgrade big sur) but  have to still using catalina end of this year -- for running project's stability )
anyway thanks to made this project. faster build speed :)

**Reply by swiftyfinch** [reply]
Thanks! 🚀

**Reply by devahmedshendy** [reply]
@swiftyfinch I am using 12.04, and  I forked the project and edited the this method name then done the install directly from my repo .. and installation worked successfully .. but with your comment, do you mean we cannot work with rugby with projects on earlier versions of Xcode?

**Reply by kfmes** [reply]
@devahmedshendy i'll try your repository, thanks. i still using xcode 12.4.

**Reply by swiftyfinch** [reply]
@devahmedshendy Hello! Feel free to open pull request to support Xcode 12.4 🚀
And also we need to add CI job for checking back compatibility with Xcode 12.4. But I can do it a little bit later.

**Reply by devahmedshendy** [reply]
@swiftyfinch 
I want to contribute, and never thought method name same as local variable is a future feature of the language and thought it was a typo. I only edited `selectedPlan(plans)` to `selectPlan(plans)`.

So, what Swift feature to read So I learn about this?

Also, I really want to be a contributor, how hard it is gonna be for the whole project if I am still using Xcode 12.04? I actually find this group discussion because yesterday I failed to run rugby successfully for a project and was going to open discussion about it.

Also, I wanted to say I want to do "add the CI job", would you guide me what to learn to be able to do this as contributor? I know about programming but a beginner in iOS and don't know enough about Xcode regarding `Rugby` project goals.

**Reply by swiftyfinch** [reply]
@devahmedshendy
You can read more here - [Creating variables that call a function of the same name](https://www.hackingwithswift.com/articles/228/whats-new-in-swift-5-4).

I think if there are two people who need support `Xcode 12.4` it can be a general issue.
So and it's easy to add support if we add a new `Xcode 12.4` job to GitHub actions.

Check out [this job](https://github.com/swiftyfinch/Rugby/blob/main/.github/workflows/regress.yml#L22).
I think we can duplicate it and use `Xcode 12.4` in the second one.

Feel free to ask more questions 🏈

**Reply by devahmedshendy** [reply]
@swiftyfinch Okay, let me finish reading/learning and get back to you with any questions.

**Comment by swiftyfinch** [comment]
@kfmes #116 Done
